crates/runtime/src/instance.rs

```diff
@@ -1015,7 +1015,7 @@ impl Instance {
             // Initialize the global before writing to it
             ptr::write(to, VMGlobalDefinition::new());
 
-            match global.initializer {
+            match module.global_initializers[def_index] {
```

Comment:
The connection between indexes of `global_initializers` and `defined_global_index` seems a little subtle to me, but maybe that's just because I'm not that used to these idioms. I think `defined_global_index` should just be subtracting off `num_imported_globals`, right? Could you use `Iterator::zip` to join the initializers to their definitions?

Feel free to leave this as is, but if you see an easy way to make this more obviously correct that'd be nice.

Reply:
Ah yes indeed this can be simplified! I've switched it to iterate over `module.global_initializers` which is more direct and clearer.

Reply:
Nice! Looks great.